File: app/javascript/channels/sensor_channel.js
import consumer from "./consumer"

$(function() {
	var chart = Chartkick.charts["chart-1"];
	$('.series').each(function(index, series) { 
		consumer.subscriptions.create({ channel: "SensorChannel", series_id: series.getAttribute("series_id") }, {
			connected() {
				console.log("connected");
			},

			disconnected() {
				console.log("disconnectd");
			},

			received(data) {
				console.log("received");
				console.log(data);
				chart.data[index].data.push(data["body"]);
				chart.updateData(chart.data);
			}
		});
	});
});

File: app/channels/sensor_channel.rb
class SensorChannel < ApplicationCable::Channel
  def subscribed
    stream_for current_series
  end

  def unsubscribed
  end

  def publish_data(data)
    broadcast_to(current_series, body: [ data["x"] || DateTime.now, data["y"] ])
    #current_series.datum.create(x: data["x"], y: data["y"])
  end

  private
    def current_series
      Series.find params[:series_id]
    end
end
